include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -754,6 +763,122 @@ class BuildCallInstance : public calyx::FuncOpPartialLoweringPattern {
   ComponentOp getCallComponent(mlir::func::CallOp callOp) const;
 };
 
+template <typename VertexType>
+class CliqueGraph {
+public:
+  CliqueGraph(
+      const std::unordered_map<VertexType, std::set<VertexType>> &adjacencyList)
```

Comment:
The only thing needs to be fixed is the data type used. I had to use standard `vector`, `map`s, etc to get it working, such as here; but I thought it'd be better to use LLVM native types such as `DenseMap`, `DenseSet`, etc. But somehow I had to make some extra work like implementing customized `getTombstoneKey` and more. Can I get some pointers on that?